src/core_plugins/kibana/public/dashboard/dashboard.js

```diff
@@ -301,12 +301,17 @@ app.directive('dashboardApp', function (Notifier, courier, AppState, timefilter,
           `${VisualizeConstants.WIZARD_STEP_1_PAGE_PATH}?${DashboardConstants.ADD_VISUALIZATION_TO_DASHBOARD_MODE_PARAM}`);
       };
 
+      const addNewSearch = function addNewSearch() {
+        kbnUrl.change('/discover');
+      };
```

Comment:
Thanks a lot for detailed explanation. I will work on these enhancements on different PR.